clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreBufferService.java

```diff
@@ -146,13 +151,29 @@ protected MemoryBoundBlockingQueue<QueueNode> getDrainerForConsumerRecord(
     return blockingQueueArr.get(drainerIndex);
   }
 
+  /**
+   * {@link #getDrainerIndexForConsumerRecord} hashes the topic name and partition to determine a drainer. Due to the
+   * different naming conventions for RT (_rt) and Separate RT (_rt_sep), different drainers might be assigned while
+   * the same drainer handling both topics would be more ideal for concurrency. Normalizing topic name fixes this issue.
+   */
+  private int computeTopicHashCode(PubSubTopic topic) {
+    if (topic.isSeparateRealTimeTopic()) {
+      return sepRtHashCodeCache.computeIfAbsent(topic, inputTopic -> {
+        String realTimeTopicName = Utils.getRealTimeTopicNameFromSeparateRealTimeTopic(inputTopic.getName());
+        PubSubTopic normalizedTopic = new PubSubTopicImpl(realTimeTopicName);
```

Comment:
can you use `this.hashCode = Objects.hash(topic, partitionNumber);` instead of creating a new `PubSubTopic` obect?